contracts/ModuleProxyFactory.sol

```diff
@@ -1,62 +1,39 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 pragma solidity >=0.8.0;
 
-// @TODO: Cesar's opinion: This is the way we should import external modules
-// currently it's not working because there are redundant declarations
-// i.e: Enum or Executor - Making the compile script to fail
-// Hence, forcing us to create the lib & interfaces folders
-
-// import "@gnosis/dao-module/contracts/DaoModule.sol";
-// import "@gnosis/AMBModule/contracts/AMBModule.sol";
-// import "@gnosis/SafeDelay/contracts/DelayModule.sol";
-
 contract ModuleProxyFactory {
-    event ModuleProxyCreation(address proxy);
-
-    uint32 public nonce = 1;
+    event ModuleProxyCreation(
+        address indexed proxy,
+        address indexed masterCopy
+    );
 
-    function createClone(address target) internal returns (address result) {
-        bytes20 targetBytes = bytes20(target);
+    function createProxy(address target, bytes32 salt)
+        internal
+        returns (address result)
+    {
+        require(
+            address(target) != address(0),
+            "createProxy: address can not be zero"
+        );
+        bytes memory deployment = abi.encodePacked(
+            hex"3d602d80600a3d3981f3363d3d373d3d3d363d73",
+            target,
+            hex"5af43d82803e903d91602b57fd5bf3"
+        );
         // solhint-disable-next-line no-inline-assembly
         assembly {
-            let clone := mload(0x40)
-            mstore(
-                clone,
-                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
-            )
-            mstore(add(clone, 0x14), targetBytes)
-            mstore(
-                add(clone, 0x28),
-                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
-            )
-            result := create(0, clone, 0x37)
+            result := create2(0, add(deployment, 0x20), mload(deployment), salt)
```

Comment:
`create2` will return the zero address in case of a failure. it might make sense to check if this is the case and revert.

Note: the only case this should revert is if the resulting address is already taken (e.g. when `deployment` and `salt` are the same for multiple calls)